Serializer/Jms/Handler/ResourceHandler.php

```diff
@@ -86,16 +86,15 @@ private function doSerializeResource(Resource $resource, $depth = 0)
 
         $data['path'] = $resource->getPath();
         $data['repository_path'] = $resource->getRepositoryPath();
-        $children = $resource->listChildren();
 
         $enhancers = $this->enhancerRegistry->getEnhancers($repositoryAlias);
 
         $children = array();
-        foreach ($resource->listChildren() as $name => $resource) {
+        foreach ($resource->listChildren() as $name => $childResource) {
```

Comment:
the `$resource` variable was already used for the main resource. This caused the enhancers to get the last child resource instead of the main resource.